cayenne-jcache/src/main/java/org/apache/cayenne/jcache/JCacheQueryCache.java

```diff
@@ -156,4 +163,37 @@ protected String cacheName(QueryMetadata metadata) {
     public void shutdown() {
         cacheManager.close();
     }
+    
+    @Override
+    public void clearLocalCache(Optional<String> namespace) {
+    	if (!namespace.isPresent()) {
+    		return;
+    	}
+
+    	for (String name : cacheManager.getCacheNames()) {
+    		Cache<String, List> cache = getCache(name);
+    		Iterator<Entry<String, List>> it = cache.iterator();
+    		while (it.hasNext()) {
+    			Entry<String, List> entry = (Entry<String, List>) it.next();
+    			if (entry.getKey().startsWith(namespace.get())) {
+    				it.remove();
+    			}
+    		}
+    	}
+    }
+
+	@Override
+	public List<String> debugListCacheKeys() {
```

Comment:
Don't think this should be part of public API, we don't provide any tools to manage cache (and we probably shouldn't). If someone need this, `javax.cache.CacheManager` can be injected and analyzed directly. Or, even better, specialized tools of actual JCache provider can be used.